### PR TITLE
test(NODE-4693): test lambda function

### PR DESCRIPTION
**Pull request by durran**
### Description

This is the Node implementation for automated testing a deployed lambda function.

#### What is changing?
- Adds the lambda function in test/lambda
- Creates the evergreen tasks to execute the scripts in drivers-evergreen-tools. (https://github.com/mongodb-labs/drivers-evergreen-tools/pull/274)

Spec: https://github.com/mongodb/specifications/blob/master/source/faas-automated-testing/faas-automated-testing.rst

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

WRITING-11435/DRIVERS-2384

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: test/lambda/events/event.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (durran) [inline test/lambda/events/event.json:1]: This is auto-generated and nothing read in the actual function from it.

**Review by W-A-James — CHANGES_REQUESTED** [review]

File: .evergreen/run-deployed-lambda-aws-tests.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (durran) [reply .evergreen/run-deployed-lambda-aws-tests.sh:1]: It looks like the AWS token expires in some cases before the script finishes: "The security token included in the request is expired". I'll get together with the build team and see if we can extend the expiration time.

File: .evergreen/run-deployed-lambda-aws-tests.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+set -o errexit  # Exit the script with error if any of the commands fail
+
+. ${DRIVERS_TOOLS}/.evergreen/run-deployed-lambda-aws-tests.sh
```
[the remark this answers is not in the document]
Comment (durran) [reply .evergreen/run-deployed-lambda-aws-tests.sh:4]: I've updated the script in tools to ensure that the entire script will run, ensuring the cluster always gets deleted: https://github.com/mongodb-labs/drivers-evergreen-tools/pull/296

[the remark this answers is not in the document]
Comment (durran) [reply .evergreen/run-deployed-lambda-aws-tests.sh:4]: That script is now merged, I've updated the PR to point back at tools master.

**Review by W-A-James — APPROVED** [review]

**Pull request merged**